inc/entityconfig.class.php

```diff
@@ -156,32 +268,48 @@ static function getUsedConfig($fieldref, $entities_id, $fieldval = '', $default_
          if ($entityConfig->getFromDB($entities_id)) {
             if (is_numeric($default_value)
                   && ($entityConfig->fields[$fieldref] != self::CONFIG_PARENT)) {
-                     return $entityConfig->fields[$fieldval];
+               return $entityConfig->fields[$fieldref];
             }
-            if (!is_numeric($default_value)) {
-               return $entityConfig->fields[$fieldval];
+            if (!is_numeric($default_value)
+                  && ($entityConfig->fields[$fieldref] != self::CONFIG_PARENT_STRING)) {
+               return $entityConfig->fields[$fieldref];
             }
 
          }
       }
 
       // Entity data not found or not defined : search in parent one
       if ($entities_id > 0) {
-
-         if ($entity->getFromDB($entities_id)) {
-            $ret = self::getUsedConfig($fieldref, $entity->fields['entities_id'], $fieldval,
-                  $default_value);
-            return $ret;
-
-         }
+         return self::getUsedConfig($fieldref, $entity->fields['entities_id'], $default_value);
       }
-      /*
-       switch ($fieldval) {
-       case "tickettype" :
-       // Default is Incident if not set
-       return Ticket::INCIDENT_TYPE;
-       }
-       */
+
       return $default_value;
    }
+
+   /**
+    * Load the plugin configuration in a global variable $PFC_CONFIG
+    *
+    * @global array $PFC_CONFIG
+    */
+   static function loadConfiguration() {
+      global $PFC_CONFIG;
+
+      // Get global configuration parameters
+      $PFC_CONFIG['tickets_summary'] = self::getUsedConfig('tickets_summary',
+         $_SESSION['glpiactive_entity']);
+      $PFC_CONFIG['user_preferences'] = self::getUsedConfig('user_preferences',
+         $_SESSION['glpiactive_entity']);
+      $PFC_CONFIG['avatar'] = self::getUsedConfig('avatar',
+         $_SESSION['glpiactive_entity']);
+      $PFC_CONFIG['user_name'] = self::getUsedConfig('user_name',
+         $_SESSION['glpiactive_entity']);
+      $PFC_CONFIG['profile_selector'] = self::getUsedConfig('profile_selector',
+         $_SESSION['glpiactive_entity']);
+      $PFC_CONFIG['external_links_prefix'] = self::getUsedConfig('external_links_prefix',
+         $_SESSION['glpiactive_entity'], self::CONFIG_PARENT_STRING);
+      $PFC_CONFIG['external_links_icon'] = self::getUsedConfig('external_links_icon',
+         $_SESSION['glpiactive_entity'], self::CONFIG_PARENT_STRING);
+      $PFC_CONFIG['external_links_title'] = self::getUsedConfig('external_links_title',
+         $_SESSION['glpiactive_entity'], self::CONFIG_PARENT_STRING);
+   }
```

Comment:
Hi

Please avoid adding a new global var. This is a bad practice. Maybe should you save this in $_SESSION['plugin_formcretor'] (which does not exists yet).

Reply:
I just fixed a typo in my comment... and you copy/pasted it ```$_SESSION['plugin_formcretor']``` is missing a A !